UnityEngine.AssetBundles/Editor/AssetBundleBrowser/InspectTab/AssetBundleInspectTab.cs

```diff
@@ -133,39 +133,128 @@ private void OnGUIEditor()
+                if (m_SelectedBundleTreeItems != null && m_SelectedBundleTreeItems.Count > 1)
+                {
+                    var style = GUI.skin.label;
+                    style.alignment = TextAnchor.MiddleCenter;
+                    style.wordWrap = true;
+                    GUI.Label(
+                    new Rect(m_Position.x + m_Position.width / 2.0f, m_Position.y + 30, m_Position.width / 2.0f, m_Position.height - 30),
+                    new GUIContent("Mutli-select inspection not supported"),
+                    style);
+                }
+                else
+                {
+                    m_SingleInspector.OnGUI(new Rect(m_Position.x + m_Position.width / 2.0f, m_Position.y + 30, m_Position.width / 2.0f, m_Position.height - 30));
+                }
             }
         }
 
-        //TODO - this is largely copied from BuildTab, should maybe be shared code.
-        private void BrowseForFolder()
+        private void RemoveSelectedItems()
         {
-            var newPath = EditorUtility.OpenFolderPanel("Bundle Folder", m_Data.m_BundlePath, string.Empty);
+            foreach(var selectedBundleTreeItem in m_SelectedBundleTreeItems)
+            {
+                m_Data.RemovePath(selectedBundleTreeItem.bundlePath);
+            }
+            RefreshBundles();
+            m_SelectedBundleTreeItems.Clear();
+        }
+
+        private void BrowseForFile()
+        {
+            var newPath = EditorUtility.OpenFilePanelWithFilters("Bundle Folder", string.Empty, new string[] { });
             if (!string.IsNullOrEmpty(newPath))
             {
                 var gamePath = System.IO.Path.GetFullPath(".");//TODO - FileUtil.GetProjectRelativePath??
                 gamePath = gamePath.Replace("\\", "/");
                 if (newPath.StartsWith(gamePath))
                     newPath = newPath.Remove(0, gamePath.Length + 1);
-                m_Data.m_BundlePath = newPath;
+
+                AddFilePath(newPath);
+
+                RefreshBundles();
             }
         }
+
+        //TODO - this is largely copied from BuildTab, should maybe be shared code.
+        private void BrowseForFolder(string folderPath = null)
+        {
+           folderPath = EditorUtility.OpenFolderPanel("Bundle Folder", string.Empty, string.Empty);
+            if (!string.IsNullOrEmpty(folderPath))
+            {
+                var gamePath = System.IO.Path.GetFullPath(".");//TODO - FileUtil.GetProjectRelativePath??
+                gamePath = gamePath.Replace("\\", "/");
+                if (folderPath.StartsWith(gamePath))
+                    folderPath = folderPath.Remove(0, gamePath.Length + 1);
+
+                AddFolderFilePath(folderPath);
+
+                RefreshBundles();
+            }
+        }
+
+        public void AddFilePath(string filePath)
+        {
+            if (m_Data.Contains(filePath))
+                return;
+
+            var bundleTestPath = this.LoadBundle(filePath);
```

Comment:
this could be a non-trivial amount of time. Especially if the bundle is LZMA compressed as the entire thing will most likely have to be loaded. My vote is to always let the user add whatever they want (explicit bundle or folder of bundles), then upon inspection, either show the current data on the right side of the window, or a message saying "invalid bundle selected". That could also free us up to make the folder code a little more inclusive (not *only* look for files paired with a .manifest) since we'd be ok with false positives.

Reply:
Makes sense, when we load a folder, just include everything? Would it be a good idea to have some extensions ignored? (.meta, .manifest, .dll, ect.)

Reply:
I would say yes to ignoring extensions in the "add folder" code, but not the "add single bundle". I've seen people with some very strange bundle naming preferences.  those three extensions you mention are probably good. maybe .cs as well, but it may be better to have too many than too few things show up.